### PR TITLE
mock: add messenger app

### PR DESCRIPTION
**Pull request by davidkostov**
Added a messenger application.

### CODE REVIEW

File: src/com/clouway/mocks/Message.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    private String title, receiver, message;
+    private Integer messageLength;
+    private MessageValidator validator = new MessageValidator();
+
+    public Message(String title, String receiver, String message) {
+        this.message = message;
+        this.title = title;
+        this.receiver = receiver;
+        messageLength = message.length();
```
Comment (mgenov) [inline src/com/clouway/mocks/Message.java]: This variable is not needed as you may take the length during the check.

**Comment by davidkostov** [comment]
Removed the variable.

File: src/com/clouway/mocks/Messenger.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.clouway.mocks;
+
+/**
+ * @author David Kostov (david.kostov.cw@gmail.com)
+ */
+public interface Messenger {
+    public boolean sendMessage(Message message);
```
Comment (mgenov) [inline src/com/clouway/mocks/Messenger.java]: public modifier is not required for interface.

File: src/com/clouway/mocks/Gateway.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private Messenger messenger;
+
+    public Gateway(Validator validator, Messenger messenger){
+        this.validator = validator;
+        this.messenger = messenger;
+    }
+
+    public boolean passMessage(Message message){
+        if(validator.isValid(message)){
+            if(messenger.sendMessage(message)){
```
Comment (mgenov) [inline src/com/clouway/mocks/Gateway.java]: return messager.sendMessage(message)

**Comment by davidkostov** [comment]
Fixed.

**Pull request merged**